crates/router/src/db/business_profile.rs

```diff
@@ -114,38 +114,89 @@ impl BusinessProfileInterface for Store {
 impl BusinessProfileInterface for MockDb {
     async fn insert_business_profile(
         &self,
-        _business_profile: business_profile::BusinessProfileNew,
+        business_profile: business_profile::BusinessProfileNew,
     ) -> CustomResult<business_profile::BusinessProfile, errors::StorageError> {
-        Err(errors::StorageError::MockDbError)?
+        let business_profile_insert = business_profile::BusinessProfile::from(business_profile);
+        self.business_profiles
+            .lock()
+            .await
+            .push(business_profile_insert.clone());
+        Ok(business_profile_insert)
     }
 
     async fn find_business_profile_by_profile_id(
         &self,
-        _profile_id: &str,
+        profile_id: &str,
     ) -> CustomResult<business_profile::BusinessProfile, errors::StorageError> {
-        Err(errors::StorageError::MockDbError)?
+        self.business_profiles
+            .lock()
+            .await
+            .iter()
+            .find(|business_profile| business_profile.profile_id == profile_id)
+            .ok_or(
+                errors::StorageError::ValueNotFound(
+                    "No business profile found for profile_id = {profile_id}".to_string(),
+                )
+                .into(),
+            )
+            .cloned()
     }
 
     async fn update_business_profile_by_profile_id(
         &self,
-        _current_state: business_profile::BusinessProfile,
-        _business_profile_update: business_profile::BusinessProfileUpdateInternal,
+        current_state: business_profile::BusinessProfile,
+        business_profile_update: business_profile::BusinessProfileUpdateInternal,
     ) -> CustomResult<business_profile::BusinessProfile, errors::StorageError> {
-        Err(errors::StorageError::MockDbError)?
+        self.business_profiles
+            .lock()
+            .await
+            .iter_mut()
+            .find(|bp| bp.profile_id == current_state.profile_id)
+            .map(|bp| {
+                let business_profile_updated =
+                    business_profile_update.apply_changeset(current_state);
+                *bp = business_profile_updated.clone();
+                business_profile_updated
+            })
+            .ok_or(
+                errors::StorageError::ValueNotFound(
+                    "No business profile found for profile_id = {profile_id}".to_string(),
```

Comment:
This seems incorrect, can you please check it.

Reply:
Sorry, I am still learning Rust, the code builds and clippy does not throw warnings. Could you explain more what seems incorrect?

Reply:
No worries, I meant error message that you have here is incorrect as `profile_id` that you have used in error message is neither a local variable nor passed in args.

```suggestion
                    "No business profile found for profile_id = {current_state.profile_id}".to_string(),
```